futures-core/src/never.rs

```diff
@@ -0,0 +1,17 @@
+//! Definition of the `Never` type,
+//! a stand-in for the `!` type until it becomes stable.
+
+/// A type with no possible values.
+///
+/// This is used to indicate values which can never be created, such as the
+/// error type of infallible futures.
+///
+/// This type is a stable equivalent to the `!` type from `std`.
+///
+/// This is currently an alias for [`std::convert::Infallible`], but in
+/// the future it may be an alias for [`!`][never].
+/// See ["Future compatibility" section of `std::convert::Infallible`][infallible] for more.
+///
+/// [never]: https://doc.rust-lang.org/nightly/std/primitive.never.html
+/// [infallible]: https://doc.rust-lang.org/nightly/std/convert/enum.Infallible.html#future-compatibility
+pub type Never = core::convert::Infallible;
```

Comment:
would it make sense to impl `std::error::Error` for this so that it can be used as a Never error type? I know so far we've used a bunch of bounds in tower like so `S::Error: Into<Box<std::error::Error>>` and this would make it compat with that.

Reply:
@LucioFranco `Infallible` does not implement `std::error::Error`, but `Into<Box<std::error::Error>>` you mentioned seems to work. https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=cda064b0d4459549673d30c79d70c886

Reply:
Awesome, thank you!

Reply:
`Infallible` seems to implement `std::error::Error` via `std::string::ParseError`.
* [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=da5cc6914193a2d97223c5b77590db87)
* [`std::string::ParseError`](https://doc.rust-lang.org/nightly/std/string/type.ParseError.html)